src/Microsoft.DotNet.Darc/src/Darc/Operations/GatherDropOperation.cs

```diff
@@ -1601,40 +1601,46 @@ private static bool IsAzureDevOpsArtifactsUrl(string location)
                     Directory.CreateDirectory(directory);
                 }
 
-                if (authHeader != null)
-                {
-                    requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri)
-                    {
-                        Headers =
-                        {
-                            { HttpRequestHeader.Authorization.ToString(), authHeader.ToString() }
-                        }
-                    };
-                }
-                else
-                {
-                    requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri);
-                }
-
                 // Ensure the parent target directory has been created.
                 using (FileStream outStream = new FileStream(temporaryFileName,
-                                                      _options.Overwrite ? FileMode.Create : FileMode.CreateNew,
-                                                      FileAccess.Write))
+                                                        _options.Overwrite ? FileMode.Create : FileMode.CreateNew,
+                                                        FileAccess.Write))
                 {
-                    using (var response = await client.SendAsync(requestMessage))
-                    {
-                        response.EnsureSuccessStatusCode();
-
-                        using (var inStream = await response.Content.ReadAsStreamAsync())
+                    await ExponentialRetry.Default.RetryAsync(
+                        async () =>
                         {
-                            downloadOutput.AppendLine($"    {sourceUri} =>");
-                            foreach (string targetFile in targetFiles)
+                            if (authHeader != null)
                             {
-                                downloadOutput.AppendLine($"      {targetFile}");
+                                requestMessage = new HttpRequestMessage(HttpMethod.Get, sourceUri)
```

Comment:
The HttpRequestMessage is IDisposable. I think we were missing the dispose call before. I think we want to wrap the whole https://github.com/dotnet/arcade-services/pull/1575/files?diff=unified&w=1#diff-66d7e4aa8669364fa4e42010f203f4fa6ddc132b6d237db44b6ed8e2bd813d3bR1612-R1640 here with try/finally and call requestMessage.Dispose() in the finally.

Reply:
The dispose is down below in the finally. Is that not enough? Is having a try inside of a try ok?

Reply:
Oh oh, missed that one.  Nested tries are fine. In your new code there's a possiblity that the request message could be constructed more than once, so it would need to be disposed each time. Nested tries are fine.

Reply:
We might create multiple requests here, and all of them need to be disposed.  The fact that exactly one needs to leak out is going to be exciting. :-)

Reply:
Based on the reading later, we probably need a "catch (HttpRequestException) { request.Dispose(); }" (since in that case, we are going to create a new one)

Reply:
If the HttpClient had a "DelegatingHandler" as it's "Handler", you could just resend the initial request without having to recreate it.  That doesn't work with the default HttpClientHandler though.

Reply:
try/finally in the inner block would pass through the finally before hitting the retry catch that exists in RetryAsync, both in success and failure cases.

Reply:
We could have a generic HttpClient that does retrying implicitly that all this code shares. So every web call everywhere doesn't have to manage this itself.

Reply:
What line is the finally on?  This block is do big (and now there are so many comments), that I can't really tell. :-)

Reply:
I don't see a try inside the callback, so I don't see how it could handle the failure cases, since the outer code has no way to detect they even happened.

Reply:
https://github.com/dotnet/arcade-services/pull/1575/files#diff-66d7e4aa8669364fa4e42010f203f4fa6ddc132b6d237db44b6ed8e2bd813d3bR1684

Reply:
So I guess I just don't understand how to have that sort of catch inside the exponential retry. I'll reach out offline to figure this out.

Reply:
> We could have a generic HttpClient that does retrying implicitly that all this code shares. So every web call everywhere doesn't have to manage this itself.

We actually do have HttpRequestManager, which we use for some Git and AzDO repo functionality. Perhaps I should just use that instead.

Reply:
I moved the requestMessage to be created in a using, which should solve the disposal issue. Let me know if I missed something.